File: Solutions/6kyu/6kyu_simple_fun_is_divisible_by_6.py
def is_divisible_by_6(s):
    if not is_even(s): return []
    return list(
             filter(checker, 
               (s.replace('*', str(n)).lstrip('0') for n in range(10))
                    )
                     )    
        

checker = lambda n: is_even(n) and sum(map(int, str(n))) % 3 == 0 if n else False    
        
is_even = lambda s: any(s[-1].endswith(i) for i in '02468*')

File: Solutions/6kyu/6kyu_simple_fun_is_divisible_by_6.js
function isDivisibleBy6(s) {
  var result = []
  if (!isEven(s)) { return result }
  
  for (var j=0; j < 10; j++) {
    var check = s.replace(/\*/, j)
    if ( checkNum(check) ) { result.push(check) }
  }
  
  return result
}


const isEven = (s, nums = ['0','2','4','6','8','*']) => nums.some(num => num == String(s).slice(-1)[0])


function checkNum(num) {
  return isEven(num) && String(num)
                        .split('')
                        .map(s => parseInt(s))
                        .reduce((x,y)=> x+y) % 3 == 0
}
